app_flutter/test/service/appengine_cocoon_test.dart

```diff
@@ -227,6 +263,34 @@ void main() {
         expect(service.rerunTask(Task(), null),
             throwsA(const TypeMatcher<AssertionError>()));
       });
+
+      /// This requires a separate test run on the web platform.
+      test('should query correct endpoint whether web or mobile', () async {
+        final Client mockClient = MockHttpClient();
+        when(mockClient.post(argThat(endsWith('/api/reset-devicelab-task')),
+                headers: captureAnyNamed('headers'),
+                body: captureAnyNamed('body')))
+            .thenAnswer((_) => Future<Response>.value(Response('', 200)));
+        service = AppEngineCocoonService(client: mockClient);
+
+        await service.rerunTask(Task(), '');
+
+        if (kIsWeb) {
+          verify(mockClient.post(
+            '/api/reset-devicelab-task',
+            headers: captureAnyNamed('headers'),
+            body: captureAnyNamed('body'),
+          ));
+        } else {
+          verify(mockClient.post(
+            'https://flutter-dashboard.appspot.com/api/reset-devicelab-task',
+            headers: captureAnyNamed('headers'),
+            body: captureAnyNamed('body'),
+          ));
+        }
+      });
     });
   });
 }
+
+class MockHttpClient extends Mock implements Client {}
```

Comment:
Looks like a good use case to put the new tests into a group: https://flutter.dev/docs/cookbook/testing/unit/introduction#5-combine-multiple-tests-in-a-group

And a group can avoid the redundant setup code. Is that right? @jonahwilliams

Reply:
Right now the tests are grouped by the endpoint they relate to. Should we swap it so we group by the functionality under test?

Reply:
Either way is fine.